### ISSUE TITLE
Issue with path resolution on Windows

### ISSUE THREAD

**Issue by piozygmunt** [typicode/husky#1397]
Hi. After upgrading to version 9 from 8 hooks on windows stopped working. Based on some quick debugging it seems that path to the actual script is not being resolved correctly. To confirm this i echo'ed script path in `h` file:

```
#!/usr/bin/env sh
echo $0
set -x
h="${0##*/}"
s="${0%/*/*}/$h"

[ ! -f "$s" ] && exit 0
...
```

which gives me this output:

```
.husky\_/pre-commit
++ h=pre-commit
++ s='.husky\_/pre-commit/pre-commit'
++ '[' '!' -f '.husky\_/pre-commit/pre-commit' ']'
++ exit 0
```

Notice that path contains both forward and back slashes and because of that `s="${0%/*/*}/$h"` returns incorrect path.
Could it be something with my setup or the regexp needs to be updated ?

OS: `Microsoft Windows [Version 10.0.19045.4046]`

**Comment by Jxlle** [comment]
+1

I have a similar problem. I use GitKraken which uses WSL to execute its git commands, but I selected the Windows installed Git executable. The following script inside all of the default hook files

```
#!/usr/bin/env sh
. "${0%/*}/h"
```

works on my Windows terminal but not through GitKraken because it uses the Windows path separator `\`. For example, the path will be `.husky\_\post-checkout` and instead of modifying it to be `.husky\_\h`, ik will be `.husky\_\post-checkout/h` and fail.

**Comment by QuantumQuin** [comment]
+1 for gitkraken issue

**Comment by DVersavel** [comment]
+ 1
Is there any news on this issue?

**Comment by sergei-lobanov** [comment]
I have issue like this, when I did upgrade from v8 to v9

According to migration guide paths to husky.sh should be removed

v8:
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

cd desktop && yarn test
```

v9:
`cd desktop && yarn test`

But if remove path, it work on MAC/Unix, but on windows I can't commit anymore

error: cannot spawn desktop/.husky/pre-commit: No such file or directory

**Comment by Jxlle** [comment]
I have created a PR that fixes the issue on my end: https://github.com/typicode/husky/pull/1402

**Comment by typicode** [comment]
@hyperupcall do you have any thoughts on this?

I find this behavior surprising as I'm not aware of other Git GUI tools on Windows having this issue?

**Comment by siddheshranade** [comment]
> But if remove path, it works on MAC/Unix, but on windows I can't commit anymore
> error: cannot spawn desktop/.husky/pre-commit: No such file or directory

I'm facing the exact same issue as @sergei-lobanov mentioned above. I'm using Windows 10.

When I restore these two script lines: 
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"
```

It works again perfectly as it did, even after keeping the change of `"prepare": "husky"` from `"prepare": "husky install"`.

Should I update to v9 with these scripts restored? Please advise.

**Comment by DVersavel** [comment]
Any news on this issues?

**Comment by pabloimrik17** [comment]
Same here on Windows 10 with Webstorm

**Comment by typicode** [comment]
Should be fixed in v9.1.0, thanks for your patience and sorry for the issues caused!
